### ISSUE TITLE
Any way to disable an inspection on a specific parent method?

### ISSUE THREAD

**Issue by Growiel** [kalessil/phpinspectionsea#1782]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) latest version |
| Language level | PHP 7.4                    |

Question
--
Hello,

Is there any way to disable an inspection on the parent level?

I have the following helper method in a Symfony project:

```
protected function getRequest(): ?Request
{
    return $this->requestStack->getCurrentRequest();
}
```

While this **could** return null, in normal uses the chances that it does are close to none, event the docs say that we can assume that the Request will be set at that point.

The issue is that every bit of code that uses this helper now gets flagged with a `[EA] Null pointer exception may occur here. `:

![image](https://user-images.githubusercontent.com/5228735/143434997-c9e63e7c-9e14-4779-a299-2823b1a49cd6.png)

I'd like to disable the inspection directly in the helper so none of the code using it gets the warning.

I don't want to disable the inspection all together since the other use cases are useful.

Thanks!

**Comment by JacekAndrzejewski** [comment]
You can disable it on the parent class level.

It works for me in this case:
```PHP
/** @noinspection NullPointerExceptionInspection */
class Req {

    public function getRequest() : ?Req
    {
        if(rand(2,3) === 3) {
            return null;
        }
        return $this;
    }

    public function getSomething()
    {
        return 4;
    }
}

class Cont {

    public function search()
    {
        $req = new Req();
        return $req->getRequest()->getSomething();
        
    }
}
```

If I remove the `@noinspection` then I get a warning in the `search` method. However I'm not 100% sure this is exactly what you want, because it will disable this inspection for all of the code in the helper and all of the calls to its methods, not only the `getRequest`.

Other solution is to override or reimplement the helper method like this:
```PHP
    protected function getRequest(): Request
    {
        $request = $this->requestStack->getCurrentRequest();
        if(!$request) {
            throw new NullRequestException();
        }
        return $request;
    }
```
If request happens to be null your code will throw an Exception even if you don't do it, but you will catch it early. And if it doesn't throw then you're guaranteed to have non null value returned.
You can do that because you can return a more specific type (Request is a subtype of ?Request) when overriding.
You can read more in [php manual](https://www.php.net/manual/en/language.oop5.variance.php).

**Comment by Growiel** [comment]
Thanks, I reworked my method to throw an exception, since this is what will happen anyways later on in the code.
